src/test/jtx/impl/Env.cpp

```diff
@@ -566,8 +558,21 @@ Env::do_rpc(
     std::vector<std::string> const& args,
     std::unordered_map<std::string, std::string> const& headers)
 {
-    return rpcClient(args, app().config(), app().logs(), apiVersion, headers)
-        .second;
+    auto response =
+        rpcClient(args, app().config(), app().logs(), apiVersion, headers);
+
+    for (unsigned ctr = 0; (ctr < retries_) and (response.first == rpcINTERNAL);
+         ++ctr)
+    {
+        JLOG(journal.error())
+            << "Env::do_rpc error, retrying, attempt #" << ctr + 1 << " ...";
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+
+        response =
+            rpcClient(args, app().config(), app().logs(), apiVersion, headers);
+    }
+
+    return response.second;
```

Comment:
I think there's a way to write this without duplicating the call to `rpcClient`.
```suggestion

    unsigned ctr = 0
    d0
    {
        auto const response =
            rpcClient(args, app().config(), app().logs(), apiVersion, headers);
        if (ctr >= retries || response.first != rpcINTERNAL)
            return response.second;

        ++ctr;
        JLOG(journal.error())
            << "Env::do_rpc error, retrying, attempt #" << ctr << " ...";
        std::this_thread::sleep_for(std::chrono::milliseconds(500));
    }
    while (true);
```
(Untested)

Reply:
It was on purpose for better understanding. From the code you will see without analyzing that rpc query will run at least once, no matter what settings used.  And `for`  is better structured than "wild"   `do/while`

Reply:
> It was on purpose for better understanding. From the code you will see without analyzing that rpc query will run at least once, no matter what settings used. And `for` is better structured than "wild" `do/while`

I want to hear more about why you call `do/while` "wild", but in this case, it's your call.